lxd/network_allocations.go

```diff
@@ -0,0 +1,237 @@
+//      type: string
+//      example: default
+//    - in: query
+//      name: all-projects
+//      description: Retrieve entities from all projects
+//      type: boolean
+//  responses:
+//    "200":
+//      description: API endpoints
+//      schema:
+//        type: object
+//        properties:
+//          type:
+//            type: string
+//            description: Response type
+//            example: sync
+//          status:
+//            type: string
+//            description: Status description
+//            example: Success
+//          status_code:
+//            type: integer
+//            description: Status code
+//            example: 200
+//          metadata:
+//            type: array
+//            description: List of network allocations used by a consuming entity
+//            items:
+//              $ref: "#/definitions/NetworkAllocations"
+//    "403":
+//      $ref: "#/responses/Forbidden"
+//    "500":
+//      $ref: "#/responses/InternalServerError"
+func networkAllocationsGet(d *Daemon, r *http.Request) response.Response {
+	projectName, _, err := project.NetworkProject(d.State().DB.Cluster, projectParam(r))
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	allProjects := shared.IsTrue(queryParam(r, "all-projects"))
+
+	var projectNames []string
+	err = d.db.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+		// Figure out the projects to retrieve.
+		if !allProjects {
+			projectNames = []string{projectName}
+		} else {
+			// Get all project names if no specific project requested.
+			projectNames, err = dbCluster.GetProjectNames(ctx, tx.Tx())
+			if err != nil {
+				return fmt.Errorf("Failed loading projects: %w", err)
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// Helper function to get the default CIDR address (/32 or /128 mask for ipv4 or ipv6 respectively)
+	getDefaultCIDRAddr := func(addr, ipv4Mask, ipv6Mask string) (string, error) {
+		ip := net.ParseIP(addr)
+		if ip.To4() != nil {
+			return fmt.Sprintf("%s%s", addr, ipv4Mask), nil
+		} else if ip.To16() != nil {
+			return fmt.Sprintf("%s%s", addr, ipv6Mask), nil
+		} else {
+			return "", fmt.Errorf("%s is not a valid IP address.", addr)
+		}
+	}
+
+	result := make([]api.NetworkAllocations, 0)
+
+	// Then, get all the networks, their network forwards and their network load balancers.
+	for _, projectName := range projectNames {
+		networkNames, err := d.db.Cluster.GetNetworks(projectName)
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		// Get all the networks, their attached instances, their network forwards and their network load balancers.
+		for _, networkName := range networkNames {
+			n, err := network.LoadByName(d.State(), projectName, networkName)
+			if err != nil {
+				return response.SmartError(err)
+			}
+
+			var ipv4Mask, ipv6Mask string
+			ipv4 := strings.Split(n.Config()["ipv4.address"], "/")
```

Comment:
No I think we just need to use the config values of ipv4.address and ipv6.address for the network addresses, and then skip the gateway entry from leases, and then use /32 or /128 for the other lease entries.

This way the network address itself will include the network-wide subnet in its cidr, but the individual leases will use singlular cidr entries.

Reply:
Lets not include them for now.